app/src/main/java/com/sdp13epfl2021/projmag/model/TagsBaseManager.kt

```diff
@@ -70,7 +70,7 @@ import java.util.regex.Pattern
                 return InputResult.AlreadyExists
             }
             else{
-                tags.add(tag)
+                tags.add(tag.copy())
```

Comment:
This is not protecting enough, because we can get all Tags, even if we copy when adding, they can be modified after.